adapter/gin/middleware.go

```diff
@@ -0,0 +1,39 @@
+package gin
+
+import (
+	"net/http"
+
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/gin-gonic/gin"
+)
+
+// SentinelMiddleware
+func SentinelMiddleware(opts ...Option) gin.HandlerFunc {
+	options := evaluateOptions(opts)
+	return func(c *gin.Context) {
+		resourceName := c.Request.Method + "_" + c.Request.URL.Path
```

Comment:
Can we use `:` as the separator here so that it could keep consistent with web adapters in other languages :)